table/format.cc

```diff
@@ -360,8 +360,13 @@ Status UncompressBlockContentsForCompressionType(
       UncompressData(uncompression_info, data, n, &uncompressed_size,
                      GetCompressFormatForVersion(format_version), allocator);
   if (!ubuf) {
+    if (!CompressionTypeSupported(uncompression_info.type()))
+      return Status::NotSupported(
+          "Unsupported compression method for this build",
+          CompressionTypeToString(uncompression_info.type()));
+
     return Status::Corruption(
```

Comment:
For code clarity, this might be better as an if-else

Reply:
That's a better style. I've updated the PR.

Reply:
@mrambacher PTAL!